File: src/app/shared/service/api/wharehouse.service.ts
import { HttpClient, HttpRequest, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { WharehouseDTORequest } from '../../model/request/wharehouseDTORequest';
import { environment } from 'src/environments/environment';
import { Observable } from 'rxjs';

@Injectable({
    providedIn: 'root'
})
export class WharehouseService {
    constructor(
        protected _httpClient: HttpClient
    ) { }

    public save(dto: WharehouseDTORequest) {
        const requ = new HttpRequest('POST', `${environment.urlBase}/wharehouses/wharehouse`, dto, {
            reportProgress: true,
            responseType: 'json'
        });
        return this._httpClient.request(requ);
    }

    public update(dto: WharehouseDTORequest, id: number) {
        const requ = new HttpRequest('PUT', `${environment.urlBase}/wharehouses/wharehouse/` + id, dto, {
            reportProgress: true,
            responseType: 'json'
        });
        return this._httpClient.request(requ);
    }

    public delete(id: number) {
        const requ = new HttpRequest('PATCH', `${environment.urlBase}/wharehouses/wharehouse/` + id, null, {
            reportProgress: true,
            responseType: 'json'
        });
        return this._httpClient.request(requ);
    }

    public findById(id: number): Observable<any> {
        return this._httpClient.get<any>(environment.urlBase + '/wharehouses/wharehouse/' + id);
    }

    public findByAllTypeDistribution(id:number): Observable<any> {
        return this._httpClient.get<any>(environment.urlBase + '/wharehouses/wharehouse/finAllDistribution/'+id);
    }

    public findAllTypeWharehouse(id:number): Observable<any> {
        return this._httpClient.get<any>(environment.urlBase + '/wharehouses/wharehouse/findAllType/'+id);
    }

    public findByName(name: string, page: number, size: number): Observable<any> {
        let queryParameters = new HttpParams();
        queryParameters = queryParameters.append('page', page);
        queryParameters = queryParameters.append('size', size);
        queryParameters = queryParameters.append('field', '');
        queryParameters = queryParameters.append('order', '');
        queryParameters = queryParameters.append('name', name);
        return this._httpClient.get<any>(environment.urlBase + '/wharehouses/wharehouse?' + queryParameters);
    }
}

File: src/main/java/com/sys/botica/crce/pe/sys_botica/service/WharehouseService.java
package com.sys.botica.crce.pe.sys_botica.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import com.sys.botica.crce.pe.sys_botica.dto.HrefEntityDTO;
import com.sys.botica.crce.pe.sys_botica.dto.WharehouseDTO;
import com.sys.botica.crce.pe.sys_botica.dto.request.WharehouseDTORequest;

public interface WharehouseService {
	public Page<WharehouseDTO> findByname(String name, Pageable pageable);
	public WharehouseDTO findById(Long id);
	public HrefEntityDTO save(WharehouseDTORequest dto);
	public HrefEntityDTO update(WharehouseDTORequest dto, Long id);
	public HrefEntityDTO delete(Long id);
	public List<WharehouseDTO> findByAllTypeDistribution();
}
